tests/gateway/test_integration.py

```diff
@@ -352,3 +381,91 @@ def _mock_request_session(max_retries, backoff_factor, retry_codes):
         requests.exceptions.HTTPError, match="Unprocessable Entity for"
     ):
         query(route=route.name, data=data)
+
+
+def test_mlflow_chat(gateway):
+    client = MlflowGatewayClient(gateway_uri=gateway.url)
+    route = client.get_route("chat-oss")
+    expected_output = {
+        "candidates": [
+            {
+                "message": {
+                    "role": "assistant",
+                    "content": "test",
+                },
+                "metadata": {"finish_reason": "length"},
+            }
+        ],
+        "metadata": {
+            "input_tokens": None,
+            "output_tokens": None,
+            "total_tokens": None,
+            "model": "mpt-chatbot",
+            "route_type": "llm/v1/chat",
+        },
+    }
+
+    data = {"messages": [{"role": "user", "content": "test"}]}
+
+    async def mock_chat(self, payload):
+        return expected_output
+
+    with patch.object(MLflowProvider, "chat", mock_chat):
+        response = client.query(route=route.name, data=data)
+    assert response == expected_output
+
+
+def test_mlflow_completions(gateway):
+    client = MlflowGatewayClient(gateway_uri=gateway.url)
+    route = client.get_route("completions-oss")
+    expected_output = {
+        "candidates": [
+            {
+                "text": "test",
+                "metadata": {"finish_reason": "length"},
+            }
+        ],
+        "metadata": {
+            "input_tokens": None,
+            "output_tokens": None,
+            "total_tokens": None,
+            "model": "mpt-completion-model",
+            "route_type": "llm/v1/completions",
+        },
+    }
+
+    data = {"prompt": "this is a test"}
+
+    async def mock_completions(self, payload):
+        return expected_output
+
+    with patch.object(MLflowProvider, "completions", mock_completions):
+        response = client.query(route=route.name, data=data)
+    assert response == expected_output
+
+
+def test_mlflow_embeddings(gateway):
+    set_gateway_uri(gateway_uri=gateway.url)
+    route = get_route("embeddings-oss")
+    expected_output = {
+        "embeddings": [
+            [0.001, -0.001],
+            [0.002, -0.002],
+        ],
+        "metadata": {
+            "input_tokens": None,
+            "output_tokens": None,
+            "total_tokens": None,
+            "model": "sentence-transformers",
+            "route_type": "llm/v1/embeddings",
+        },
+    }
+
+    data = {"text": ["test1", "test2"]}
+
+    async def mock_embeddings(self, payload):
+        return expected_output
+
+    with patch.object(MLflowProvider, "embeddings", mock_embeddings):
```

Comment:
```suggestion
    with patch.object(MLflowProvider, "embeddings", return_value=...):
```

Can we use `return_value`?